src/test/java/com/iabtcf/encoder/field/FixedVectorEncoderTest.java

```diff
@@ -0,0 +1,32 @@
+package com.iabtcf.encoder.field;
+
+import com.iabtcf.model.SortedVector;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FixedVectorEncoderTest {
+    private static final String VALID_VALUE = "1010101000";
+    private static final SortedVector EXPECTED_RETURN = new SortedVector();
+    private static final String INVALID_VALUE = null;
+
+    @Before
+    public void before() {
+        for (int i = 0; i < VALID_VALUE.length(); i++) {
+            if (VALID_VALUE.charAt(i) == '1')
+                EXPECTED_RETURN.getSet().add(i + 1);
+        }
+        EXPECTED_RETURN.setBitLength(VALID_VALUE.length());
+    }
+
+    @Test
+    public void whenValueIsValidThenShouldReturnExpectedSortedVector() {
+        Assert.assertEquals(EXPECTED_RETURN, FixedVectorEncoder.getInstance().decode(VALID_VALUE));
+    }
+
+    @Test
+    public void whenValueIsInvalidThenShouldReturnNull() {
+        Assert.assertNull(FixedVectorEncoder.getInstance().decode(INVALID_VALUE));
+    }
+
```

Comment:
How does the class under test behave if you fill a string not containing numeric values, say for example "abc102(&$_-"?

Reply:
The suggested change was made.